src/axom/inlet/examples/mfem_coefficient.cpp

```diff
@@ -0,0 +1,190 @@
+                       "The function representing the BC coefficient");
+
+    schema.addFunction("coef",
+                       inlet::FunctionType::Double,
+                       {inlet::FunctionType::Vec3D},
+                       "The function representing the BC coefficient");
+
+    schema.addFunction("vec_coef_t",
+                       inlet::FunctionType::Vec3D,
+                       {inlet::FunctionType::Vec3D,
+                        inlet::FunctionType::Double},  // Multiple argument types
+                       "The function representing the BC coefficient");
+
+    schema.addFunction("coef_t",
+                       inlet::FunctionType::Double,
+                       {inlet::FunctionType::Vec3D, inlet::FunctionType::Double},
+                       "The function representing the BC coefficient");
+  }
+};
+
+/**
+ * Example Lua definition:
+ * \code{.lua}
+ * bc = {
+ *   attrs = {
+ *      3, 4, 6, 9
+ *   }
+ *   coef = function (x, y, z)
+ *     return x * 0.12
+ *   end
+ * }
+ * -- or, for vector coefficients:
+ * [8] = {
+ *   attrs = { [4] = 14, [8] = 62, [6] = 11},
+ *   vec_coef = function (x, y, z)
+ *     scale = 0.12
+ *     return x * scale, y * scale, z * scale
+ *   end
+ * }
+ * \endcode
+ */
+template <>
+struct FromInlet<BoundaryCondition>
+{
+  BoundaryCondition operator()(const inlet::Table& base)
+  {
+    BoundaryCondition bc;
+    bc.attrs = base["attrs"];
+    if(base.contains("vec_coef"))
+    {
+      // We assume the dimension is 3 - if the mfem::Vector is 2D,
+      // the z-component will be set to zero due to how primal::Vector's
+      // (T*, size) constructor works
+      bc.vec_coef = std::make_unique<mfem::VectorFunctionCoefficient>(
+        3,
+        [&base](const mfem::Vector& input, mfem::Vector& output) {
+          // One way of accessing the function is with "call" - need
+          // to specify the desired return and argument types explicitly
+          auto ret = base["vec_coef"].call<axom::primal::Vector3D>(
+            axom::primal::Vector3D {input.GetData(), input.Size()});
+          // Copy from the primal vector into the MFEM vector
+          std::copy(ret.data(), ret.data() + ret.dimension(), output.GetData());
+        });
+      SLIC_INFO("Created an mfem::VectorCoefficient with dimension "
+                << bc.vec_coef->GetVDim());
+    }
+    else if(base.contains("coef"))
+    {
+      // Another way of accessing the function is by extracting the std::function
+      auto func =
+        base["coef"].get<std::function<double(axom::primal::Vector3D)>>();
+      bc.coef = std::make_unique<mfem::FunctionCoefficient>(
+        [func(std::move(func))](const mfem::Vector& vec) {
+          // func is a concrete function type so calls can leverage conversions
+          return func({vec.GetData(), vec.Size()});
+        });
+      SLIC_INFO("Created an mfem::Coefficient");
+    }
+    else if(base.contains("coef_t"))
```

Comment:
It may be cleaner to just always assume these are time varying. I think it would reduce duplication.